subsys/net/lib/config/init.c

```diff
@@ -563,14 +590,502 @@ int net_config_init_app(const struct device *dev, const char *app_info)
 	return ret;
 }
 
+static struct net_if *get_interface(const struct net_init_config *config,
+				    int config_ifindex,
+				    const struct device *dev,
+				    const char **iface_name)
+{
+	const struct net_init_config_iface *cfg;
+	struct net_if *iface = NULL;
+	const char *name;
+
+	NET_ASSERT(IN_RANGE(config_ifindex, 0, ARRAY_SIZE(config->iface) - 1));
+
+	cfg = &config->iface[config_ifindex];
+
+	name = cfg->new_name;
+	if (name != NULL) {
+		iface = net_if_get_by_index(net_if_get_by_name(name));
+	}
+
+	if (iface == NULL) {
+		name = cfg->name;
+
+		if (name != NULL) {
+			iface = net_if_get_by_index(net_if_get_by_name(name));
+		}
+	}
+
+	if (iface == NULL) {
+		/* Get the interface by device */
+		const struct device *iface_dev;
+
+		name = cfg->dev;
+
+		iface_dev = device_get_binding(name);
+		if (iface_dev) {
+			iface = net_if_lookup_by_dev(iface_dev);
+		}
+	}
+
+	if (iface_name != NULL) {
+		*iface_name = name;
+	}
+
+	return iface;
+}
+
+static void ipv6_setup_by_yaml(struct net_if *iface,
```

Comment:
Perhaps "yaml" word should be ditched from the code here as it does not do anything with yaml syntax anywhere. (It's a bit misleading, at first look I was looking for some actual yaml parsing somewhere...)
Maybe ipv6_setup_from_generated?

Reply:
> Maybe ipv6_setup_from_generated?

Good point, there is nothing yaml specific here, I will fix this in next version.